src/roman_datamodels/datamodels/_utils.py

```diff
@@ -24,7 +34,84 @@ class FilenameMismatchWarning(UserWarning):
     """
 
 
-def _node_update(to_node, from_node, extras=None, extras_key=None, ignore=None):
+def _temporary_update(
+    datamodel: DataModel, key: str, value: Any, wrapper: type[TaggedScalarNode] | None = None
+) -> Generator[None, None, None]:
+    """
+    Temporary update some meta key of a datamodel so that it can be saved with
+    that value without changing the current model's version of that value.
+
+    Parameters
+    ----------
+    datamodel :
+        The datamodel instance to update.
+
+    key : str
+        The key to update in the datamodel's meta attribute.
+
+    value : Any
+        The value to set for the key in the datamodel's meta attribute.
+
+    wrapper : type[TaggedScalarNode] | None
+        If provided, the value will be wrapped in this type before being set.
+        This is useful for ensuring that the value is of the correct type.
+    """
+    if "meta" in datamodel._instance and key in datamodel._instance.meta:
+        old_value = getattr(datamodel._instance.meta, key)
+        setattr(datamodel._instance.meta, key, wrapper(value) if wrapper else value)
```

Comment:
Is this wrapper needed? I thought assigning to `meta.filename` and `meta.file_date` will already [convert the value to the required node](https://github.com/spacetelescope/roman_datamodels/blob/f1d126fc23af643f0aaa4f6496057055fe4cdbeb/src/roman_datamodels/stnode/_node.py#L55).